src/main/java/edu/kit/ipd/crowdcontrol/objectservice/event/EventObservable.java

```diff
@@ -25,16 +27,25 @@ public EventObservable() {
     /**
      * Will emit the given object as event to the subscribed subscribers of the observable.
      * @param t The object to pass to the events
+     * @return The messages returned by the emits
      */
-    public void emit(T t) {
-        subs.stream().forEach(subscriber -> subscriber.onNext(t));
+    public List<EventLog> emit(T t) {
+        return subs.stream()
+                .map(subscriber -> eventEmit(subscriber, t))
+                .collect(Collectors.toList());
```

Comment:
this is sequential, i don't think we want this to be sequential. you can use parStream() to achieve concurrency